src/react/databrowser/objects/MetadataSearch.jsx

```diff
@@ -69,13 +73,19 @@ const MetadataSearch = ({
       return;
     }
     dispatch(newSearchListing(bucketName, inputText));
+    // Add metadatasearch in the query params
+    query.set('metadatasearch', inputText);
+    dispatch(push(`${pathname}?${query.toString()}`));
   };
 
   const reset = e => {
     e.stopPropagation();
     setHintsShown(false);
     setInputText('');
     dispatch(listObjects(bucketName, prefixWithSlash));
```

Comment:
I think this line may be dropped ?

Reply:
🤔  I think if we really want to drop this line, then we probably need to add the value of metadatasearch query param in the dependency list where we dispatch the `listObjects`. 

I am wondering, it is something we want?

```js
  useEffect(() => {
    dispatch(
      listObjects(
        bucketNameParam,
        prefixWithSlash,
        isShowVersions ? LIST_OBJECT_VERSIONS_S3_TYPE : LIST_OBJECTS_S3_TYPE,
      ),
    ).finally(() => setLoaded(true));
  }, [bucketNameParam, prefixWithSlash, dispatch, isShowVersions]);
  ```

Reply:
OK. So we took the decision to centralize the listObjects only in one place (Objects.jsx)